torchrl/collectors/collectors.py

```diff
@@ -278,6 +296,22 @@ def __init__(
                     )
                 env.update_kwargs(create_env_kwargs)
 
+        if passing_device is None:
+            if device is not None:
+                passing_device = device
+            elif policy is not None:
+                try:
+                    policy_device = next(
+                        policy.parameters()
+                    ).device  # Get the first device from the policy parameter list
+                except:  # noqa
```

Comment:
What kind of exception is this? I believe we can use StopIteration or something like that for the `except`